llap-tez/src/java/org/apache/hadoop/hive/llap/tezplugins/LlapTaskSchedulerService.java

```diff
@@ -790,6 +794,8 @@ public void run() {
         }
       }, 0, 10000L, TimeUnit.MILLISECONDS);
 
+      llapMetricsCollector.start();
```

Comment:
Should we maybe move the start below the code, adding the collector to the registry? Otherwise, the collector doesn't have any LLAP daemons and the first scheduler execution is a NOOP....

Reply:
Now this is a bit tricky. I didn't wanted to do a tight coupling between the registry and the collector, but I agree that there is a dependency between the starts. So I've added the ServiceStateChangeListener to the Collector, and subscribed to the state changes of the registry. This way we can ensure, that the starts and stops happen in the correct way.